packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableRowAction.tsx

```diff
@@ -1,69 +1,105 @@
-import { Popover } from '@headlessui/react'
-import { EllipsisVerticalIcon } from '@heroicons/react/24/outline'
-import dayjs from 'dayjs'
-import { twMerge } from 'tailwind-merge'
-import { useMemo } from 'react'
+import { useCallback, useMemo } from 'react'
 import { GET_HELP_LINK } from '../../constants'
 import { useClaimWithdrawal } from '../../hooks/useClaimWithdrawal'
-import { MergedTransaction } from '../../state/app/state'
+import { DepositStatus, MergedTransaction } from '../../state/app/state'
 import { useClaimCctp, useRemainingTime } from '../../state/cctpState'
 import { trackEvent } from '../../util/AnalyticsUtils'
 import { isUserRejectedError } from '../../util/isUserRejectedError'
 import { getNetworkName } from '../../util/networks'
 import { errorToast } from '../common/atoms/Toast'
 import { Button } from '../common/Button'
-import { Tooltip } from '../common/Tooltip'
 import { useSwitchNetworkWithConfig } from '../../hooks/useSwitchNetworkWithConfig'
 import { useNetwork } from 'wagmi'
-
-const GetHelpButton = ({
-  variant,
-  onClick
-}: {
-  variant: 'primary' | 'secondary'
-  onClick: () => void
-}) => {
-  return (
-    <Button
-      variant={variant}
-      onClick={onClick}
-      className={variant === 'secondary' ? 'bg-white px-4 py-3' : ''}
-    >
-      Get Help
-    </Button>
-  )
-}
+import { isDepositReadyToRedeem } from '../../state/app/utils'
+import { useRedeemRetryable } from '../../hooks/useRedeemRetryable'
+import { WithdrawalCountdown } from '../common/WithdrawalCountdown'
+import { DepositCountdown } from '../common/DepositCountdown'
+import { Address } from '../../util/AddressUtils'
 
 export function TransactionsTableRowAction({
   tx,
   isError,
-  type
+  type,
+  address
 }: {
   tx: MergedTransaction
   isError: boolean
   type: 'deposits' | 'withdrawals'
+  address: Address | undefined
 }) {
   const { chain } = useNetwork()
-  const { switchNetwork } = useSwitchNetworkWithConfig()
+  const { switchNetworkAsync } = useSwitchNetworkWithConfig()
   const networkName = getNetworkName(chain?.id ?? 0)
 
-  const { claim, isClaiming } = useClaimWithdrawal()
+  const { claim, isClaiming } = useClaimWithdrawal(tx)
   const { claim: claimCctp, isClaiming: isClaimingCctp } = useClaimCctp(tx)
-  const { isConfirmed } = useRemainingTime(tx)
+  const { redeem, isRedeeming } = useRedeemRetryable(tx, address)
+  const { remainingTime: cctpRemainingTime } = useRemainingTime(tx)
+
+  const isConnectedToCorrectNetworkForClaim = useMemo(() => {
+    if (!chain) {
+      return false
+    }
+    return chain.id === tx.destinationChainId
+  }, [chain, tx.destinationChainId])
 
-  const currentChainIsValid = useMemo(() => {
+  const isConnectedToCorrectNetworkForRedeem = useMemo(() => {
     if (!chain) {
       return false
     }
-    if (type === 'deposits') {
-      return chain.id === tx.childChainId
+    return chain.id === tx.childChainId
+  }, [chain, tx.childChainId])
```

Comment:
Can we use optional chaining here? So we could get rid of `useMemo` at the same time (same for `isConnectedToCorrectNetworkForClaim`

`const isConnectedToCorrectNetworkForRedeem = chain?.id === tx.childChainId`